lib/src/rules/prefer_relative_imports.dart

```diff
@@ -93,34 +75,19 @@ class _Visitor extends SimpleAstVisitor<void> {
       return false;
     }
 
-    // Is the package: import referencing the current package?
-    var segments = uri.pathSegments;
-    if (segments.isEmpty) return false;
-
-    if (parsedPubspec == null) {
-      String content;
-      try {
-        content = pubspecFile.readAsStringSync();
-        // ignore: avoid_catches_without_on_clauses
-      } catch (_) {}
-      parsedPubspec = _parseYaml(content);
-    }
-
-    return parsedPubspec['name'] == segments[0];
+    final source = node.uriElement?.library?.source;
+    if (source == null) return false;
+    // todo (pq): should context.package.contains(source)work?
```

Comment:
Yes, that should work.

Reply:
Oddly it's not...  I'll look a little closer.

/fyi @srawlins